### PR TITLE
Add env variable for init_on_device

### PR DESCRIPTION
**Pull request by shingjan**
Fixes [#25448](https://github.com/huggingface/transformers/issues/25448) from transformers

cc: @sgugger

### CODE REVIEW

File: src/accelerate/big_modeling.py [hunk 1]
```diff
@@ -100,6 +101,7 @@ def init_on_device(device: torch.device, include_buffers: bool = False):
     ```
     """
     # TODO(shingjan): remove the torch version check once older versions are deprecated
+    include_buffers = parse_flag_from_env('INIT_INCLUDE_BUFFERS', False)
```
Comment (sgugger) [inline src/accelerate/big_modeling.py:104]: Let's just set the default (at line 83) to `None` and then do this only if `include_buffers` is `None` (so the user can override the default from the env variable).

Also please leave the comment after this addition (before the torch version test).

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! Left a few comments.

File: src/accelerate/big_modeling.py (hunk 1 above)
Comment (muellerzr) [reply src/accelerate/big_modeling.py:104]: Let's also add `ACCELERATE_` as a prefix please so we can keep consistency with the rest of the library :)

Comment (shingjan) [reply src/accelerate/big_modeling.py]: Thanks @sgugger and @muellerzr for the review! I also make the same change to default value of `include_buffers` for `init_empty_weights` to make the env variable useful and keep BC. Thanks!

File: src/accelerate/big_modeling.py [hunk 2]
```diff
@@ -47,13 +48,13 @@
...
 @contextmanager
-def init_empty_weights(include_buffers: bool = False):
+def init_empty_weights(include_buffers: bool = None):
     """
     A context manager under which models are initialized with all parameters on the meta device, therefore creating an
     empty model. Useful when just initializing the model would blow the available RAM.
 
     Args:
-        include_buffers (`bool`, *optional*, defaults to `False`):
+        include_buffers (`bool`, *optional*, defaults to `None`):
```
Comment (sgugger) [inline src/accelerate/big_modeling.py:57]:
```suggestion
        include_buffers (`bool`, *optional*):
```
Small nit: we don't put the default when it's None as it's implied by `*optional*`

File: src/accelerate/big_modeling.py [hunk 3]
```diff
@@ -74,19 +75,21 @@ def init_empty_weights(include_buffers: bool = False):
...
-def init_on_device(device: torch.device, include_buffers: bool = False):
+def init_on_device(device: torch.device, include_buffers: bool = None):
     """
     A context manager under which models are initialized with all parameters on the specified device.
 
     Args:
         device (`torch.device`):
             Device to initialize all parameters on.
-        include_buffers (`bool`, *optional*, defaults to `False`):
+        include_buffers (`bool`, *optional*, defaults to `None`):
```
Comment (sgugger) [inline src/accelerate/big_modeling.py:92]:
```suggestion
        include_buffers (`bool`, *optional*):
```

**Review by sgugger — APPROVED** [review]
Thanks for iterating! Two last small suggestions and we should be ready to go.

**Comment by shingjan** [comment]
thanks @sgugger and @muellerzr for getting this in!

**Pull request merged**